CMakeLists.txt

```diff
@@ -69,108 +58,101 @@ if (CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
           set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -march=native")
         endif()
     endif()
+
     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -Wextra -Werror")
     set(EXTRA_CXX_FLAGS -Weffc++ -Wswitch-default -Wfloat-equal -Wconversion -Wsign-conversion)
-    if (RAPIDJSON_BUILD_CXX11 AND CMAKE_VERSION VERSION_LESS 3.1)
-        if (CMAKE_CXX_COMPILER_VERSION VERSION_LESS "4.7.0")
-            set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++0x")
-        else()
-            set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11")
-        endif()
-    endif()
-    if (RAPIDJSON_BUILD_ASAN)
-        if (CMAKE_CXX_COMPILER_VERSION VERSION_LESS "4.8.0")
+
+    if(RAPIDJSON_BUILD_ASAN)
+        if(CMAKE_CXX_COMPILER_VERSION VERSION_LESS "4.8.0")
             message(FATAL_ERROR "GCC < 4.8 doesn't support the address sanitizer")
         else()
             set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fsanitize=address")
         endif()
     endif()
-    if (RAPIDJSON_BUILD_UBSAN)
-        if (CMAKE_CXX_COMPILER_VERSION VERSION_LESS "4.9.0")
+
+    if(RAPIDJSON_BUILD_UBSAN)
+        if(CMAKE_CXX_COMPILER_VERSION VERSION_LESS "4.9.0")
             message(FATAL_ERROR "GCC < 4.9 doesn't support the undefined behavior sanitizer")
         else()
             set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fsanitize=undefined")
         endif()
     endif()
-elseif (CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+elseif(CMAKE_CXX_COMPILER_ID MATCHES "Clang")
     if(NOT CMAKE_CROSSCOMPILING)
```

Comment:
This has been there before, but the check isn't necessary: ```CMAKE_SYSTEM_PROCESSOR``` is about the target system, so the check should be fine.

Reply:
this CXX_FLAGS setting is not well at all!